workflow/controller/exec_control.go

```diff
@@ -21,7 +21,11 @@ func (woc *wfOperationCtx) applyExecutionControl(ctx context.Context, pod *apiv1
 	}
 
 	nodeID := woc.nodeID(pod)
-
+	node := woc.wf.Status.Nodes[nodeID]
```

Comment:
No able to reproduce on my local env. Looks like some edge case pod state and node status is out of sync and it try to update the completed node

Reply:
Added extra check to make sure node status is not fulfilled for update the status